docs: add required API key headers for 1.5 (#8721)
langflow-ai/langflow: docs/docs/Configuration/configuration-api-keys.md

import TabItem from '@theme/TabItem';
import Icon from "@site/src/components/icon";

You can use Langflow API keys to interact with Langflow programmatically.

> The API key has the same permissions and access as you do when logged in through the web interface. This means your API key can only access your own flows, components, and data, and cannot access other users' resources.

Reviewer: ```suggestion
The API key has the same permissions and access as you do when you launch Langflow. This means your API key can only access your own flows, components, and data. You cannot access other users' resources with your own Langflow API keys.
```

The API key represents the user who created them. If you create your key as a superuser, you will have superuser privileges and can manage other users, but will do so with your API key.

<details open>
<summary>Auto-login and API key authentication</summary>

> Prior to Langflow v1.5, when `AUTO_LOGIN` was enabled with `AUTO_LOGIN=true`, Langflow automatically logged users in as a superuser without requiring authentication, and API requests could be made without a Langflow API key.

Reviewer 2: .

As of Langflow v1.5, all API requests require a Langflow API key, even when `AUTO_LOGIN` is enabled.

The only exceptions are the MCP endpoints at `/v1/mcp`, `/v1/mcp-projects`, and `/v2/mcp`.
The MCP-related endpoints will continue to require no authentication when `AUTO_LOGIN=true`.

</details>

## Generate a Langflow API key

> You can generate a Langflow API key with the UI, or the CLI.

Reviewer: Noob question: Is the CLI installed automatically when you install LF?

> The UI-generated key is appropriate for most cases. The CLI key is needed when your Langflow server is running in `--backend-only` mode.

Reviewer 2: The CLI-generated?

Author: Good suggestion!

<Tabs>
  <TabItem value="Langflow UI" label="Langflow UI" default>

1. Click your user icon, and then select **Settings**.
2. Click **Langflow API Keys**, and then click **Add New**.
3. Name your key, and then click **Create API Key**.
4. Copy the API key and store it in a secure location.

  </TabItem>

  <TabItem value="Langflow CLI" label="Langflow CLI">

    If you're serving your flow with `--backend-only=true`, you don't have a way to create an API key within the UI.

> To create API keys with the Langflow CLI, `AUTO_LOGIN` must be set to `FALSE` and you must be logged in as a superuser.

Reviewer 2: Hm, this seems weird. Why do we do this?

Author: @Cristhianzl question for you

To create an API key for a user from the CLI, do the following:

1. In your `.env` file, set `AUTO_LOGIN=FALSE`, and set superuser credentials for your server.

    ```text
    LANGFLOW_AUTO_LOGIN=False
    LANGFLOW_SUPERUSER=administrator
    LANGFLOW_SUPERUSER_PASSWORD=securepassword
    ```

> 2. To confirm your superuser status, check that `is_superuser` is `true` in the response from the [users/whoami](/api-users#get-current-user) endpoint.

Reviewer: ```suggestion
2. To confirm your superuser status, call [`GET /users/whoami`](/api-users#get-current-user), and then check that the response contains `"is_superuser": true`:
```

    ```bash
    curl -X GET \
      "$LANGFLOW_URL/api/v1/users/whoami" \
      -H "accept: application/json" \
      -H "x-api-key: $LANGFLOW_API_KEY"
    ```

    <details closed>
    <summary>Result</summary>

    ```json
    {
      "id": "07e5b864-e367-4f52-b647-a48035ae7e5e",
      "username": "langflow",
      "profile_image": null,
      "store_api_key": null,
      "is_active": true,
      "is_superuser": true,
      "create_at": "2025-05-08T17:59:07.855965",
      "updated_at": "2025-05-29T15:06:56.157860",
      "last_login_at": "2025-05-29T15:06:56.157016",
    }
    ```

    </details>

2. Create an API key:

    ```shell
    uv run langflow api-key
    ```
  </TabItem>
</Tabs>

## Authenticate requests with the Langflow API key

Include your API key in API requests to authenticate requests to Langflow.

API keys allow access only to the flows and components of the specific user to whom the key was issued.

<Tabs>
  <TabItem value="HTTP header" label="HTTP header" default>

To use the API key when making API requests, include the API key in the HTTP header:

```shell
curl -X POST \

>   "http://LANGFLOW_SERVER_ADDRESS/api/v1/run/FLOW_ID?stream=false" \

Reviewer 2: $

Reviewer 2: (same comment elsewhere throughout)

  -H 'Content-Type: application/json' \

>   -H 'x-api-key: LANGFLOW_API_KEY' \

Reviewer 2: These all also need $, right?

You can use Langflow API keys to interact with Langflow programmatically.

The API key has the same permissions and access as you do when you launch Langflow. This means your API key can only access your own flows, components, and data. You cannot access other users' resources with your own Langflow API keys.
An API key represents the user who created it. If you create a key as a superuser, then that key will have superuser privileges.
Anyone who has that key can authorize superuser actions through the Langflow API, including user management and flow management.

In Langflow versions 1.5 and later, most API requests require a Langflow API key, even when `AUTO_LOGIN=true`.

The only exceptions are the MCP endpoints: `/v1/mcp`, `/v1/mcp-projects`, and `/v2/mcp`.
These endpoints don't require authentication, regardless of the `AUTO_LOGIN` setting.

<details>
<summary>Auto-login and API key authentication in earlier Langflow versions</summary>

If you are running a Langflow version earlier than 1.5, if `AUTO_LOGIN=true`, Langflow automatically logs users in as a superuser without requiring authentication, and API requests can be made without a Langflow API key.

> If you set `SKIP_AUTH_AUTO_LOGIN=true`, authentication will be skipped entirely, and API requests will not require a Langflow API key, regardless of the `AUTO_LOGIN` setting.

Author: ```suggestion
If you set `SKIP_AUTH_AUTO_LOGIN=true` and `AUTO_LOGIN=true`, authentication will be skipped entirely, and API requests will not require a Langflow API key.
```

import TabItem from '@theme/TabItem';
import Icon from "@site/src/components/icon";

> Langflow provides an API key functionality that allows users to access their individual components and flows.

Reviewer: This intro is a bit esoteric. Can we make it more layperson-friendly? Like: "You can use Langflow API keys to interact with Langflow programmatically, including accessing your custom components and editing flows".

This would be a good place to explain how the API key acts (whether it takes on the role of the user who created it or if it's more like a literal representation of the user's account).

Author: Added

## Auto-login and API key authentication

> Prior to Langflow v1.5, when `AUTO_LOGIN` was enabled with `AUTO_LOGIN=true`, Langflow automatically logged users in as a superuser without requiring authentication, and API requests could be made without a Langflow API key.

Reviewer: I think it would be clearer to separate the version-specific info on tabs, or put the "pre-1.5" info in a `<details>`.

For example, how I show the Java driver dependency examples for different versions:
https://docs.datastax.com/en/datastax-drivers/compatibility/java-drivers.html#java-driver-ownership

As of Langflow v1.5, all API requests require a Langflow API key, even when `AUTO_LOGIN` is enabled.

> The MCP-related endpoints will continue to require no authentication when `AUTO_LOGIN=true`.

Reviewer: These two sentences contradict each other. Is it all endpoints or all-except-MCP? and which are the MCP endpoints?

> ### Generate an API key with the Langflow UI

Reviewer: This would be a good spot for tabs (put the UI and CLI options in tabs).

To create an API key for a user from the CLI, do the following:

1. In your `.env` file, set `AUTO_LOGIN=FALSE`, and set superuser credentials for your server.

    ```text
    LANGFLOW_AUTO_LOGIN=False
    LANGFLOW_SUPERUSER=administrator
    LANGFLOW_SUPERUSER_PASSWORD=securepassword
    ```

2. To confirm your superuser status, check that `is_superuser` is `true` in the response from the [users/whoami](/api-users#get-current-user) endpoint.

    ```bash
    curl -X GET \
      "$LANGFLOW_URL/api/v1/users/whoami" \
      -H "accept: application/json" \
      -H "x-api-key: $LANGFLOW_API_KEY"
    ```

    Result:
    ```json
    {
      "id": "07e5b864-e367-4f52-b647-a48035ae7e5e",
      "username": "langflow",
      "profile_image": null,
      "store_api_key": null,
      "is_active": true,
      "is_superuser": true,
      "create_at": "2025-05-08T17:59:07.855965",
      "updated_at": "2025-05-29T15:06:56.157860",
      "last_login_at": "2025-05-29T15:06:56.157016",
    }
    ```

2. To create an API key, run:

    ```shell
    uv run langflow api-key
    ```

> ## Authenticate requests with the Langflow API key

Reviewer: Another good spot for tabs (Header vs query parameter)
